lib/src/components/NxModal/NxModal.tsx

```diff
@@ -4,22 +4,41 @@
  * the terms of the Eclipse Public License 2.0 which accompanies this
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
-import React, {FunctionComponent} from 'react';
+import React, {FunctionComponent, useEffect} from 'react';
 import classnames from 'classnames';
 
-import {Props, propTypes} from './types';
+import {CloseHandler, Props, propTypes} from './types';
 
 import './NxModal.scss';
 
-const NxModal: FunctionComponent<Props> = ({className, ...attrs}) => {
+const currentModalCloseHandlers: CloseHandler[] = [];
 
+const NxModal: FunctionComponent<Props> = ({className, onClose, ...attrs}) => {
   const modalClasses = classnames('nx-modal', className);
+
+  const modalCloseListener = ({ key }: { key: string}) => {
+    if (key === 'Escape' && currentModalCloseHandlers.length
+        && onClose === currentModalCloseHandlers[currentModalCloseHandlers.length - 1]) {
+      onClose();
+    }
+  };
+
+  useEffect(function() {
+    document.addEventListener('keydown', modalCloseListener);
+
+    currentModalCloseHandlers.push(onClose);
+
+    return function() {
```

Comment:
Give a name to this function  — it'll be easier to identify in case an error ever shows up.

Reply:
> Give a name to this function — it'll be easier to identify in case an error ever shows up.

Named on bf2a3cf

Reply:
> > Give a name to this function — it'll be easier to identify in case an error ever shows up.
> 
> Named on [bf2a3cf](https://github.com/sonatype/sonatype-react-shared-components/commit/bf2a3cf46face787fbfc956b2057abe89243d65f)

You could've named it inline:

```
useEffect(() => {
    ...
    return function cleanup() { ... }
});
```